pkg/api/common/versions.go

```diff
@@ -68,11 +68,20 @@ func GetDefaultKubernetesVersionWindows() string {
 }
 
 // GetSupportedKubernetesVersion verifies that a passed-in version string is supported, or returns a default version string if not
-func GetSupportedKubernetesVersion(version string) string {
-	if k8sVersion := version; AllKubernetesSupportedVersions[k8sVersion] {
-		return k8sVersion
+func GetSupportedKubernetesVersion(version string, hasWindows bool) string {
```

Comment:
we shouldn't silently override. If a value is passed in we should throw an error if invalid.

Reply:
I would tend to agree with you but this PR is intended to fix the bug, not change the behavior of default k8s version definition... The silent override was already in place for getting the k8s version for API version `v20170701` so I don't think we want to change the behavior in this PR.

Reply:
does validation happen before this so we don't have to worry about it?

Reply:
We do validate beforehand. So if I pass in something random as `orchestratorVersion` such as `0.5` I get `FATA[0000] error loading API model in generateCmd: error parsing the api model: OrchestratorProfile has unknown orchestrator version: 0.5`
The only case where this function overrides the version is if it is empty or if the cluster is a Windows cluster and the passed in version is valid but not supported for Windows, in which case the user gets default. I looked into fixing validation to also take into account "is this a Windows cluster? If so, is the passed in version valid for Windows?" but it's not straight forward since it requires having the agent pool profiles. I can make that an additional change in this PR if we see value in it.
TLDR: Is it okay to override the version if the value passed in for the user is not supported with windows but is a valid version or should we fail instead?